guides/development-workflow.md

```diff
@@ -0,0 +1,341 @@
+
+To deploy:
+
+1. review what work is currently awaiting deployment by looking at Trello
+2. let the team know you're deploying to Production
+3. deploy
+
+```
+git checkout origin/master
+git merge origin/develop
+git push
+```
+
+4. verify the deployment has been successful
+5. update Trello with all the cards that have now been deployed
+
+---
+
+## Branching
+
+#### How to make a branch
+
+Create the branch off of `develop`:
+
+```
+git fetch -a
+git checkout develop
+git branch <branch-name>
+git checkout <branch-name>
+
+# or the shorthand
+
+git checkout -b <branch-name> origin/develop
+```
+
+**Naming the branch**
+
+Having a convention allows for:
+
+* all branches connected to a story or ticket are easily connected now and in the future
+* others can easily identify the intent of the contents without having to interpret a commit's title
+* encourages small and concise commits which keeps the focus on the type of change being made. If you are adding a new feature and fix something along the way, that *should* be 2 commits instead of 1
+
+**Convention:**
+
+* Feature - A branch that adds a new feature, as defined by the specified story
+* Fix - A branch that corrects a problem in a feature already merged into `develop`
+* Chore - Chore branches are used for routine tasks or tickets which are not emergencies
+* Hotfix - A branch that adds an urgent fix to a problem that affects production. These branches are based on `master` and do not go via `develop`, so must be also be merged into `develop` when deployed
+
+```
+feature/{STORY ID}-{CONCISE TITLE}
+fix/{STORY/TICKET ID}-{CONCISE TITLE}
+chore/{STORY ID}-{CONCISE TITLE}
+hotfix/{TICKET ID}-{CONCISE TITLE}
+```
+
+**Examples:**
+
+```
+feature/523797477-add-logging-to-registration
+fix/423797477-logging-happens-in-both-environments
+chore/1934729239-reduce-caching-for-contact-details
+hotfix/253947623-remove-breaking-change-to-repair-creation
+```
+
+#### Update your branch with the latest changes on develop (rebasing)
+
+When working in a team other pull requests will often get merged in before yours is finished  It's important to make sure you're developing against the latest version of the code base as much as possible.
+
+You can do this by rebasing your work on top of the `develop` branch.
+
+```
+git checkout <branch-name>
+git rebase origin/develop
+git push --force-with-lease
+```
+
+During the rebase there may be conflicts which should be resolved at this point. If in doubt talk to the other developer who you're conflicting with and reason out what the correct state should be.
+
+Forcing pushing in this situation is the only time it should be necessary, adding `--force-with-lease` ensures you don't accidentally overwrite someone else's additions.
+
+## Committing
+
+Writing a good commit message serves three important purposes.
+
+* to speed up the review process
+* to encourage individual commits to be well small, bounded and concise
```

Comment:
Fixed the grammar. 

I think we all agree that lots of small commits each focused on small bits of functionality is preferable over fewer large commits that do multiple things.

With that in mind when writing commit messages if you find yourself writing something like this, it's a flag that it's too big and needs to be broken down:

```
(feature): Application now does foo, bar, baz
```